### ISSUE TITLE
Warn when a computed field overwrites a reactive field

### ISSUE THREAD

**Issue by thelinuxlich** [pmndrs/eslint-plugin-valtio#5]
```js
const state = proxyWithComputed({
   counter: 0
},{
  counter(snap) { 
      // should be invalid
  }
})
```

**Comment by Aslemammad** [comment]
Could you elaborate on the example?

**Comment by thelinuxlich** [comment]
Yeah, this state definition is wrong because both the first object and the second have the same property key so this will silently fail when you try to use a snapshot.

**Comment by dai-shi** [comment]
Yeah, I noticed this limitation while developing it. I suppose we can check this in `proxyWithComputed` and ideally we should improve typing to avoid this.
